craft_cli/dispatcher.py

```diff
@@ -245,6 +245,11 @@ def load_command(self, app_config: Any) -> BaseCommand:
         emit.trace(f"Command parsed sysargs: {self._parsed_command_args}")
         return self._loaded_command
 
+    @property
+    def parsed_args(self) -> argparse.Namespace:
+        """The map of parsed command-line arguments."""
+        return self._parsed_command_args or argparse.Namespace()
```

Comment:
I also recognise that this changes the behaviour from craft-application's implementation and that I also didn't ask about this when it was added to craft-application. As such, I'd like to also check with @cmatsuoka for objections, just in case he considered and rejected that.

Reply:
![](https://y.yarn.co/707261c7-7539-4f83-9100-23f1fb96f036_text.gif)

I'm leaving this conversation open so Claudio can see my question, just in case he does have objections.

Reply:
Thanks Alex! I think this was added to make the linters happy, raising the exception is the way to go.